### PR TITLE
update rabit to v0.0.0-p1

### PR DESCRIPTION
**Pull request by headupinclouds**
add dmlc::dmlc dependency

### CODE REVIEW

File: cmake/projects/rabit/hunter.cmake [hunk 1]
```diff
@@ -14,11 +14,11 @@ hunter_add_version(
     PACKAGE_NAME
     rabit
     VERSION
-    0.0.0-p0
```
Comment (ruslo) [inline cmake/projects/rabit/hunter.cmake]: Please keep old version.

Comment (headupinclouds) [reply cmake/projects/rabit/hunter.cmake]: Ok.  I thought it would be better to remove broken releases, but I can see the problems this could cause

Comment (headupinclouds) [reply cmake/projects/rabit/hunter.cmake]: Done.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.3205
* https://travis-ci.org/ingenue/hunter/builds/326412673?utm_source=github_status&utm_medium=notification

Note that there is no Android testing at this moment, we need API 24 toolchain:
* https://github.com/ruslo/hunter/pull/1295#issuecomment-355883801

Let me know if you want me to add one.

**Comment by headupinclouds** [comment]
> Let me know if you want me to add one.

Sure, let's get Android working with the default build.

I've sent https://github.com/ruslo/polly/pull/222

Thanks.

**Comment by ruslo** [comment]
> let's get Android working with the default build

Will create minimal version of Android NDK for CI.

**Comment by ruslo** [comment]
> Will create minimal version of Android NDK for CI.

It's possible to use `android-ndk-r16b-api-24-armeabi-v7a-neon-clang-libcxx` toolchain in CI now.

GTest example, works fine:
* https://travis-ci.org/ingenue/hunter/builds/326473152?utm_source=github_status&utm_medium=notification

Rabit however still produce error:
* https://travis-ci.org/ingenue/hunter/builds/326473764?utm_source=github_status&utm_medium=notification

```
/home/travis/build/ingenue/hunter/_testing/Hunter/_Base/21ed09c/f309420/f1fa033/Build/rabit/Source/src/./socket.h:460:12: error: static_cast from 'const fd_set *' to 'fd_set *', which are not related by inheritance, is not allowed
    return FD_ISSET(fd, &read_set) != 0;
           ^~~~~~~~~~~~~~~~~~~~~~~
/home/travis/build/ingenue/hunter/_ci/android-ndk-r16b/sysroot/usr/include/sys/select.h:76:29: note: expanded from macro 'FD_ISSET'
#define FD_ISSET(fd, set) ((__FDS_BITS(set)[__FDELT(fd)] & __FDMASK(fd)) != 0)
                            ^~~~~~~~~~~~~~~
/home/travis/build/ingenue/hunter/_ci/android-ndk-r16b/sysroot/usr/include/sys/select.h:50:40: note: expanded from macro '__FDS_BITS'
#define __FDS_BITS(set) (__BIONIC_CAST(static_cast, fd_set*, set)->fds_bits)
                         ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/travis/build/ingenue/hunter/_ci/android-ndk-r16b/sysroot/usr/include/sys/cdefs.h:77:34: note: expanded from macro '__BIONIC_CAST'
#define __BIONIC_CAST(_k,_t,_v) (_k<_t>(_v))
                                 ^~~~~~~~~~
```

Not sure what it means, Android NDK issue?

**Comment by headupinclouds** [comment]
> Rabit however still produce error:

Thanks for adding that.

> error: static_cast from 'const fd_set *' to 'fd_set *', which are not related by inheritance, is not allowed

I think it just needs a `const_cast<fd_set *>(X)` somewhere (or C equivalent).  I can look later tonight.

**Comment by headupinclouds** [comment]
I've updated the rabit release.  It needed a few `const_cast<fd_set *>(X)`'s.

**Comment by ruslo** [comment]
Testing:
* https://travis-ci.org/ingenue/hunter/builds/326675830?utm_source=github_status&utm_medium=notification
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.3208

Android API 24:
* https://travis-ci.org/ingenue/hunter/builds/326676280?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.3209
* https://travis-ci.org/ingenue/hunter/builds/326679192?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.19.223

**Comment by headupinclouds** [comment]
Thanks.

**Pull request merged**